async_service/trio.py

```diff
@@ -257,8 +256,28 @@ def cancel(self) -> None:
                         )
                         self.cancel()
                         raise DaemonTaskExit(f"Daemon task {name} exited")
+                    while True:
+                        children_still_running = tuple(
```

Comment:
Can't we refactor this and move it into base.py? It seems pretty much identical to the one in asyncio